src/txn/txn_ckpt.c

```diff
@@ -1354,7 +1365,7 @@ __checkpoint_tree(
 	 * checkpoint is created, and then things get bad.
 	 */
 	WT_ERR(__wt_page_modify_init(session, btree->root.page));
-	__wt_page_modify_set(session, btree->root.page);
+	__wt_page_only_modify_set(session, btree->root.page);
```

Comment:
Please add a comment explaining why we don't want to make the tree modified here.